File: 1_Calorie_Counting/solution.py
# Part 1
calories = [0]

with open("./1_Calorie_Counting/input.txt", "r") as f:
    for line in f:
        if line.strip() == "":
            calories.append(0)
            continue
        calories[-1] += int(line)

print(f"Maximum calories: {max(calories)}")

# Part 2
top_three = sorted(calories, reverse=True)[:3]
print(f"Sum of top three: {sum(top_three)}")

File: 1_Calorie_Counting/solution.cpp
// read input.txt and add up each line. If there is an empty line append a 0

#include <iostream>
#include <fstream>
#include <string>
#include <list>
#include <sstream>
using namespace std;

int main()
{
    ifstream in("input.txt");
    string line;
    std::list<int> listOfInts;
    listOfInts.push_back(0);
    while (getline(in, line))
    {
        if (line.empty())
        {
            listOfInts.push_back(0);
        }
        else
        {
            listOfInts.back() += stoi(line);
        }
    }

    // print max value in list
    cout << *max_element(listOfInts.begin(), listOfInts.end()) << endl;

    // print sum of the three largest values in list
    listOfInts.sort();
    listOfInts.reverse();
    int sum = 0;
    for (int i = 0; i < 3; i++)
    {
        sum += listOfInts.front();
        listOfInts.pop_front();
    }
    cout << sum << endl;

    return 0;
}